src/SignalR/clients/csharp/Client.Core/src/HubConnection.cs

```diff
@@ -757,6 +757,7 @@ private async Task CommonStreaming(ConnectionState connectionState, string strea
             // It's safe to access connectionState.UploadStreamToken as we still have the connection lock
             _state.AssertInConnectionLock();
             var cts = CancellationTokenSource.CreateLinkedTokenSource(connectionState.UploadStreamToken, token);
+            // Don't run any async code before this line, we're assuming the lock is still held by the calling code, but it wont be if we go async
```

Comment:
This looks scary. Should the assertion be moved later? I would rather put code that needs to be locked in a non-async method that calls into this.

Reply:
Agreed, moved the token creation earlier to avoid this odd code.